File: DEBERES/Deberes IF y SWITCH/01-IFCondicional.ts
/*
Este Programa analiza si una función cuadratica tiene soluciones reales o imaginarias.
*/
console.log("\tBienvenido,este programa determina si la ec. cuadratica tiene soluciones reales.");
console.log("A continuacion introduzca a, b y c.\n de la forma ax^2+bx+c=0");
const variableAString = prompt("Introduce 'a': ");
const variableBString = prompt("Introduce 'b': ");
const variableCString = prompt("Introduce 'c': ");
const varA = Number(variableAString);
const varB = Number(variableBString); // Ingresan variables y las cambio a numeros
const varC = Number(variableCString);

const discriminante = Number(varB*varB-4*varA*varC); // dicriminante para sabes si es positivo o negativo
const raiz = Math.sqrt(discriminante); // funcion para sacar la raiz
const solucion1 = Number((-varB+raiz)/2*varA);  
const solucion2 = Number((-varB-raiz)/2*varA);   // soluciones de la ecuacion cuadratica
const solReales = discriminante >=0;

// Ingresan condicionales
if(solReales){
    if(discriminante>0){
    console.log("Esta ecuacion tiene dos soluciones reales, las cuales son: ");
    console.log("x= " + solucion1.toPrecision(4) + "\n" + "x= " + solucion2.toPrecision(4)); 
    }

    if(discriminante == 0){
    console.log("Esta ecuacion solo tiene una solucion real y es: ");
    console.log("x= " + solucion1 );
    }

}else{
    console.log("Esta ecuacion NO tiene soluciones Reales.");
    }

File: DEBERES/Deberes IF y SWITCH/01-IFCondicional.cpp
#include <iostream>
#include <stdio.h>
#include <math.h>
using namespace std;

int main(){
    system("color 02");
    cout<<"\tBienvenido,este programa determina si la ec. cuadratica tiene soluciones reales."<<endl;
cout<<"A continuacion introduzca a, b y c.\n de la forma ax^2+bx+c=0\n"<<endl;
float varA,varB,varC;

cout<<"Introduce 'a': "; cin>>varA;
cout<<"Introduce 'b': "; cin>>varB;
cout<<"Introduce 'c': "; cin>>varC;
cout<<endl;

float discriminante = (varB*varB-4*varA*varC); // dicriminante para sabes si es positivo o negativo
float raiz = sqrt(discriminante); // funcion para sacar la raiz
float solucion1 = ((-varB+raiz)/2*varA); 
float solucion2 = ((-varB-raiz)/(2*varA));   // soluciones de la ecuacion cuadratica
float solReales = discriminante >=0;

// Ingresan condicionales
if(solReales){
    if(discriminante>0){
    cout<<"Esta ecuacion tiene dos soluciones reales, las cuales son: "<<endl;
    cout<<"x= "<<solucion1<<endl<<"x= "<<solucion2<<endl;
    }

    if(discriminante == 0){
    cout<<"Esta ecuacion solo tiene una solucion real y es: "<<endl;
    cout<<"x= "<<solucion1<<endl;
    }

}else{
    cout<<"Esta ecuacion NO tiene soluciones Reales.";
    }

    return 0;
}
